spec/mock-request.js

```diff
@@ -1,14 +1,26 @@
 "use strict";
 const q = require("q");
+let clock = jasmine.Clock;
+let fakeDate;
+let callbacks = require("../lib/realtime-callbacks");
+
+let tick = function tick(millis) {
+    // make sure we tick the fakedate first, otherwise nothing will happen!
+    fakeDate += millis;
+    clock.tick(millis);
+};
+
 
 /**
  * Construct a mock HTTP backend, heavily inspired by Angular.js.
  * @constructor
  */
-function HttpBackend() {
+function HttpBackend(useMockClock) {
     this.requests = [];
     this.expectedRequests = [];
+    this.useMockClock = useMockClock ? true : false
```

Comment:
Ideally mock-requests should always use a mock-clock but that seems to be break some internal setTimeout uses?